src/components/FormRegister.vue

```diff
@@ -0,0 +1,221 @@
+
+    const isValid = (val: string): boolean => val?.length > 0;
+
+    const isEmail = (value: string): boolean => {
+      /**
+       * Match 99% of valid email addresses and will not pass validation
+       * for email addresses that have, for instance
+       * - Dots in the beginning
+       * - Multiple dots at the end
+       * But at the same time it will allow part after @ to be IP address.
+       *
+       * https://uibakery.io/regex-library/email
+       */
+      const regex =
+        /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/;
+      return regex.test(value);
+    };
+
+    const isStrong = (value: string): boolean => {
+      let isLong = value.length > 5;
+      let hasLetter = /[a-zA-Z]/.test(value);
+      return isLong && hasLetter;
+    };
+
+    const isIdentical = (confirm: string, password: string): boolean => {
+      return confirm === password;
+    };
+
+    const onSubmitRegister = () => {
+      // noop
+    };
+
+    return {
+      formRegister,
+      isPassword,
+      isPasswordConfirm,
+      isValid,
+      isEmail,
+      isStrong,
+      isIdentical,
+      onSubmitRegister,
+    };
+  },
+});
+</script>
+
+<template>
+  <div class="text-grey-10" data-cy="form-register">
+    <!-- Heading -->
+    <div class="q-my-lg">
+      <h1
+        class="text-h5 text-grey-10 text-bold q-my-none"
+        data-cy="form-register-title"
+      >
+        {{ $t('register.form.titleRegister') }}
+      </h1>
+    </div>
+    <!-- Form: register -->
+    <q-form @submit.prevent="onSubmitRegister">
+      <!-- Input: email -->
+      <div data-cy="form-register-email">
+        <!-- Label -->
+        <label for="form-register-email" class="text-caption text-bold">
+          {{ $t('register.form.labelEmail') }}
+        </label>
+        <!-- Input -->
+        <q-input
+          dense
+          outlined
+          v-model="formRegister.email"
+          lazy-rules
+          :rules="[
+            (val) => isValid(val) || $t('register.form.messageEmailReqired'),
+            (val) => isEmail(val) || $t('register.form.messageEmailInvalid'),
+          ]"
+          bg-color="grey-1"
+          id="form-register-email"
+          name="subject"
+          class="q-mt-sm"
+          data-cy="form-register-email-input"
+        />
+      </div>
+      <!-- Input: password -->
+      <div data-cy="form-register-password">
+        <!-- Label -->
+        <label for="form-register-password" class="text-caption text-bold">
+          {{ $t('register.form.labelPassword') }}
+        </label>
+        <!-- Input -->
+        <q-input
+          dense
+          outlined
+          hide-bottom-space
+          bg-color="grey-1"
+          v-model="formRegister.password"
+          id="form-register-password"
+          :hint="$t('register.form.hintPassword')"
+          :type="isPassword ? 'password' : 'text'"
```

Comment:
Maybe we can use `lazy-rules` validation props.

Reply:
Added.